### ISSUE TITLE
Implement jumps: H, M, L

### ISSUE THREAD

**Issue by shirshov** [nosami/XSVim#55]
TLDR;:

`H`igh, `M`iddle, `L`ow -- move the cursor to first, middle, or last visible line

From VIM docs:

H			To line [count] from top (Home) of window (default:
			first line on the window) on the first non-blank
			character |linewise|.  See also 'startofline' option.
			Cursor is adjusted for 'scrolloff' option.

M			To Middle line of window, on the first non-blank
			character |linewise|.  See also 'startofline' option.

L			To line [count] from bottom of window (default: Last
			line on the window) on the first non-blank character
			|linewise|.  See also 'startofline' option.
			Cursor is adjusted for 'scrolloff' option.

http://vimdoc.sourceforge.net/htmldoc/motion.html#jump-motions

**Comment by nosami** [comment]
I would love to have this, but it might not be so easy. Lots of editor code was made internal between XS 6.0 and VS for Mac including the current scroll position.

I'm not sure if it's possible to determine the top visible line (without resorting to reflection hacks)

**Comment by shirshov** [comment]
Looks like the correct way of getting the first/last lines is through `editor.TextView.TextViewLines.FirstVisibleLine`. This is not implemented yet:
```
System.NotImplementedException: The method or operation is not implemented.
  at Microsoft.VisualStudio.Text.Editor.Implementation.TextView.get_TextViewLines () [0x00000] in /Users/builder/data/lanes/4862/445a7f09/source/monodevelop/main/src/core/Mono.TextEditor.Platform/Platform/Text/Impl/MdView/TextView.cs:346
 ```

Source is here: https://github.com/mono/monodevelop/blob/master/main/src/core/Mono.TextEditor.Platform/Platform/Text/Impl/MdView/TextView.cs#L346

But, line 271 of TextView.cs contains this code, we should watch for it to be enabled, I guess:

```
#if TARGET_VS
        public ITextViewLineCollection TextViewLines
        {
            get
            {
                return _textViewLinesCollection;
            }
        }
#endif
```

**Comment by nosami** [comment]
doesn't look like that is defined anywhere https://github.com/mono/monodevelop/search?utf8=%E2%9C%93&q=TARGET_VS&type=
